File: Easy/1137. N-th Tribonacci Number/main.py
class Solution:
    def tribonacci(self, n: int) -> int:
        match n: # Edge cases
            case 0: return 0
            case 1 | 2: return 1 # case 1 or 2
        
        T_first, T_mid, T_last = 0, 1, 1
        for i in range(n - 2):
            T_next = T_first + T_mid + T_last
            T_first, T_mid, T_last = T_mid, T_last, T_next # [0 (first), 1 (mid), 2 (last)] 3 (next) -> 0 [1 (first), 2 (mid), 3 (last)]
        return T_next

File: Easy/1137. N-th Tribonacci Number/main.cpp
class Solution {
public:
    int tribonacci(int n) {
        switch (n) {
            case 0:
                return 0;
            case 1:
            case 2:
                return 1;
        }
        int T_first = 0, T_mid = 1, T_last = 1;
        int T_next;
        for (int i=0; i < n-2; i++) {
            T_next = T_first + T_mid + T_last;
            T_first = T_mid, T_mid = T_last, T_last = T_next;
        }
        return T_next;
    }
};
